ext/standard/basic_functions.stub.php

```diff
@@ -395,3 +395,199 @@ function assert($assertion, $description = null): bool {}
 
 /** @return int|string|bool|null */
 function assert_options(int $what, $value = UNKNOWN) {}
+
+/* string.c */
+
+function bin2hex(string $data): string {}
+
+/** @return string|false */
+function hex2bin(string $data) {}
+
+/** @return int|false */
+function strspn(string $str, string $mask, int $start = UNKNOWN, int $len = UNKNOWN) {}
+
+/** @return int|false */
+function strcspn(string $str, string $mask, int $start = UNKNOWN, int $len = UNKNOWN) {}
+
+#if HAVE_NL_LANGINFO
+/** @return string|false */
+function nl_langinfo(int $item) {}
+#endif
+
+function strcoll(string $str1, string $str2): int {}
+
+function trim(string $str, string $character_mask = " \n\r\t\v\0"): string {}
+
+function rtrim(string $str, string $character_mask = " \n\r\t\v\0"): string {}
+
+function ltrim(string $str, string $character_mask = " \n\r\t\v\0"): string {}
+
+function wordwrap(string $str, int $width = 75, string $break = "\n", bool $cut = false): string {}
+
+function explode(string $separator, string $str, int $limit = PHP_INT_MAX): array {}
+
+/**
+ * @param string $glue Optional - defaults to empty string
+ * @param array $pieces
+ */
+function implode($glue, $pieces): string {}
+
+/**
+ * @param string $str Optional - defaults to previous string
+ * @param string $token
+ * @return string|false
+ */
+function strtok($str, $token) {}
+
+function strtoupper(string $str): string {}
+
+function strtolower(string $str): string {}
+
+function basename(string $path, string $suffix = ""): string {}
+
+function dirname(string $path, int $levels = 1): string {}
+
+/** @return array|string */
+function pathinfo(string $path, int $options = UNKNOWN) {}
+
+/** @return string|false */
+function stristr(string $haystack, string $needle, bool $part = false) {}
```

Comment:
```suggestion
function stristr(string $haystack, string $needle, bool $before_needle = false) {}
```

The manual name is a lot more obvious here.